src/app/clusters/on-off-server/on-off-server.cpp

```diff
@@ -64,11 +64,38 @@ using namespace chip;
 using namespace chip::app::Clusters;
 using namespace chip::app::Clusters::OnOff;
 
-#ifdef ZCL_USING_ON_OFF_CLUSTER_START_UP_ON_OFF_ATTRIBUTE
-static bool areStartUpOnOffServerAttributesTokenized(EndpointId endpoint);
-#endif
+/**********************************************************
+ * Attributes Definition
+ *********************************************************/
+
+static std::array<OnOffEffect *, EMBER_AF_ON_OFF_CLUSTER_SERVER_ENDPOINT_COUNT> instances = { 0 };
```

Comment:
This does not work with dynamic endpoints...

Reply:
Just to follow up: on Slack I suggested that OnOffEffect can form a (doubly?) linked list, so we can have however many of them consumers want to instantiate without having to maintain a dynamic container ourselves or have some sort of max-sized array.

Reply:
PR was opened #12635